File: app/src/main/java/com/example/pba_greenspots/MiPerfil.java
package com.example.pba_greenspots;

import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

public class MiPerfil extends Fragment {

    private MiPerfilViewModel mViewModel;
    private View v;
    EditText nombre;
    EditText apellido;
    EditText pais;
    EditText email;
    EditText contrasenia;
    Button btnVolver;
    public static MiPerfil newInstance() {
        return new MiPerfil();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        v = inflater.inflate(R.layout.fragment_registro, container, false);
        nombre = v.findViewById(R.id.nombre_perfil);
        apellido = v.findViewById(R.id.apellido_perfil);
        email = v.findViewById(R.id.email_perfil);
        contrasenia = v.findViewById(R.id.contrasenia_perfil);
        btnVolver = v.findViewById(R.id.btn_volver);
        return v;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = new ViewModelProvider(this).get(MiPerfilViewModel.class);
        // TODO: Use the ViewModel
    }

}

File: app/src/main/java/com/example/pba_greenspots/fragments/MiPerfil.kt
package com.example.pba_greenspots.fragments

import androidx.lifecycle.ViewModelProvider
import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import com.example.pba_greenspots.R
import com.example.pba_greenspots.models.Usuario
import com.google.firebase.firestore.ktx.firestore
import com.google.firebase.firestore.ktx.toObject
import com.google.firebase.ktx.Firebase

class MiPerfil : Fragment() {

    companion object {
        fun newInstance() = MiPerfil()
    }

    private lateinit var viewModel: MiPerfilViewModel
    lateinit var v : View
    lateinit var nombre : EditText

    //acceso a la base de datos
    val db = Firebase.firestore


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
         v = inflater.inflate(R.layout.fragment_mi_perfil, container, false)
        nombre = v.findViewById(R.id.campoNombre)

        return v
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        viewModel = ViewModelProvider(this).get(MiPerfilViewModel::class.java)
        // TODO: Use the ViewModel
    }

    override fun onStart() {
        super.onStart()
    /*
        var mascota : Mascota = Mascota("Pedro",Mascota.Constants.typePerro,"Colie",2,"imagen.com")

        db.collection("mascotas").document(mascota.nombre).set(mascota)
        db.collection("mascotas").add(mascota)

       viewModel.initTestList()

        for (mascota in viewModel.mascotas) {
            db.collection("mascotas").document(mascota.nombre).set(mascota)
        }
  */


// Leer datos una sola vez
        var docRef = db.collection("users").document("prueba")

        docRef.get()
            .addOnSuccessListener { dataSnapshot ->
                if (dataSnapshot != null) {
                    val usuario  = dataSnapshot.toObject<Usuario>()
                    Log.d("Test", "DocumentSnapshot data: ${usuario.toString()}")
                } else {
                    Log.d("Test", "No such document")
                }
            }
            .addOnFailureListener { exception ->
                Log.d("Test", "get failed with ", exception)
            }




    }


}